.github/workflows/weekly.yml

```diff
@@ -96,3 +96,24 @@ jobs:
           name: job-results-without-plugins
           path: /home/runner/avocado/job-results/
       - run: echo "🥑 This job's status is ${{ job.status }}."
+
+
+  full-bandit-checks:
+    name: Bandit checks
+    runs-on: ubuntu-20.04
+    steps:
+      - run: echo "Job triggered by a ${{ github.event_name }} event on branch is ${{ github.ref }} in repository is ${{ github.repository }}, runner on ${{ runner.os }}"
+      - name: Check out repository code
+        uses: actions/checkout@v2
+        with:
+          fetch-depth: 0
+      - name: Run bandit check
+        run: make -f Makefile.gh bandit
```

Comment:
@beraldoleal it's doable but I find that too complicated for little benefit. Personally, I'd rather add the command line directly in the weekly github action.

Reply:
>  it's doable but I find that too complicated for little benefit. Personally, I'd rather add the command line directly in the weekly github action.

I have the hardship to find this "too difficult."  This was what I had in mind:

```bash
if [ -z "${AVOCADO_BANDIT_FULL}" ]; then
    echo "Running bandit in full mode"
    bandit -o bandit-output.txt -r --skip B101,B105,B311,B404,B603 .
else
    echo "Running bandit in high-severity mode"
    bandit -r -lll .
fi
```

Something like this would concentrate all bandit logic in a single file and inside "selftests" directory since this is a "selftest". This will make it easier for new contributors to understand the "bandit modes" we are running. 

The way it is in this PR, we have a shell script called "selftests/check-bandit.sh" giving the impression that all bandit checks related are there. If someone is reading only the `check-bandit.sh` or `Makefile.gh`, they will not be aware of all different "modes" we have.

Reply:
Regardless of having a Makefile or not, I'm still standing with my argument of concentrating this all inside "selftests/check-bandit.sh".

Reply:
So far we have:

* all files that match `selftests/*.sh` being run on every `make check`
* all CI actions that are *not* part of `make check` being put into a `Makefile`

I'm also bothered by having different commands at different places, but this PR is coherent with the logic described above.

Reply:
Sorry for insisting on this, but I'm still not convinced that the logic described is enough.

As suggested, moving the actual bandit execution inside `selftests/bandit-check.sh` would keep the logic above and also solve the part that bothers us. IMO having only those two bullets as "accepted logic" is risky because it leaves room for dispersal executions in different files, making code maintenance and reading difficult.

Currently we are "starting loosing" this: right now, in order to know what is being skipped or how bandit is being executed, I have to look at two files (Makefile.gh and selftests/check-bandint.sh). And even those are executed in different contexts, they are both "selftests".

Also, `requirements-dev.txt` has a specific bandit version while `Makefile.gh` is installing just "bandit", leaving room for inconsistent bandit versions being installed.

Reply:
bandit in makefile.gh could go away because once we have this checking in every PR the weekly check should go away.

Beraldo, I'm sorry you're not convinced, but eventually you need to evaluate if blocking something that can help the quality of our code is worthwhile so much discussion. I'm pretty sure all this part of the code (checks) is going to evolve in the next year.

Reply:
(I know the PR is closed, but just in case of a future v2)

> bandit in makefile.gh could go away because once we have this checking in every PR the weekly check should go away.

I'm reviewing PRs as it is. And whenever possible, I like to avoid thinking about future changes because I have learned the hard way that technical debits are easily forgotten, especially those "easy and for internal use only".

> Beraldo, I'm sorry you're not convinced, but eventually you need to evaluate if blocking something that can help the quality of our code is worthwhile so much discussion. I'm pretty sure all this part of the code (checks) is going to evolve in the next year.

Part of the submission process is not only convincing reviewers that your idea is good but also that the location, names, and design decisions are also good for the project. It is a slow process, and I understand that sometimes it is frustrating for who is submitting, but I like to think it is healthier in the end. I'm not saying that we can't change our minds and refactor later; of course, we can (I "unblocked" many PRs, after a good argument). I'm just saying that if we keep going in "this mode," we will quickly increase coupling and lose cohesion inside the project. We could avoid that.

I'm not buying the argument that a feature should be accepted without a proper discussion about where it should be placed just because the feature itself improves code quality.